lib/current-context.js

```diff
@@ -0,0 +1,77 @@
+// Copyright IBM Corp. 2015,2016. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var juggler = require('loopback-datasource-juggler');
+var remoting = require('strong-remoting');
+var ClsContext = require('loopback-context-cls');
+var deprecated = require('depd')('loopback');
+
+module.exports = function(loopback) {
+
+  /**
+   * Get the current context object. The context is preserved
+   * across async calls, it behaves like a thread-local storage.
+   *
+   * @returns {ChainedContext} The context object or null.
+   */
+  loopback.getCurrentContext = function() {
+    // NOTE(bajtos) ClsContext.getCurrentContext is overriden whenever
+    // the context changes, therefore we cannot simply assign
+    // ClsContext.getCurrentContext() to loopback.getCurrentContext()
+    deprecated('loopback.getCurrentContext() is deprecated. ' +
+      'Consider using different means of passing the context around. ' +
+      'As a short-term fix, use loopback-context-cls module.');
```

Comment:
Can we link to a github issue or wiki (or other doc) so that we can keep the info up to date?

A deprecation warning will be sufficient to direct the person to a doc with some concrete info. We can also keep this up to date as we look into this further.